sklearn/preprocessing/_polynomial.py

```diff
@@ -674,6 +674,9 @@ def get_feature_names_out(self, input_features=None):
             Transformed feature names.
         """
         n_splines = self.bsplines_[0].c.shape[0]
+        if self.extrapolation == "periodic":
+            n_splines -= self.degree
```

Comment:
Maybe add a comment like in the `fit`method. E.g.
`# periodic splines have self.degree less degrees of freedom`